core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -1046,18 +1055,30 @@ private[spark] object Utils extends Logging {
       return cached
     }
 
-    val indx: Int = hostPort.lastIndexOf(':')
-    // This is potentially broken - when dealing with ipv6 addresses for example, sigh ...
-    // but then hadoop does not support ipv6 right now.
-    // For now, we assume that if port exists, then it is valid - not check if it is an int > 0
-    if (-1 == indx) {
+    def setDefaultPortValue: (String, Int) = {
       val retval = (hostPort, 0)
       hostPortParseResults.put(hostPort, retval)
-      return retval
+      retval
+    }
+
+    if (hostPort != null && hostPort.split(":").length > 2) {
```

Comment:
Since the new changed logic (line 1064 ~ 1081), could you add a comment before this line?